osu.Game/Screens/Play/HUD/FailingLayer.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics;
+
+namespace osu.Game.Screens.Play.HUD
+{
+    /// <summary>
+    /// An overlay layer on top of the playfield which fades to red when the current player health falls below a certain threshold defined by <see cref="LowHealthThreshold"/>.
+    /// </summary>
+    public class FailingLayer : HealthDisplay
+    {
+        private const float max_alpha = 0.4f;
+
+        private readonly Box box;
+
+        /// <summary>
+        /// The threshold under which the current player life should be considered low and the layer should start fading in.
+        /// </summary>
+        public double LowHealthThreshold = 0.20f;
```

Comment:
Public fields are sort of eh for me. Would just rather have it a property but it's not anything major.

Reply:
Yeah, it should have stayed an auto-prop but CI inspection complains about it being convertible to an get-only auto-prop so I made it a field to make CI stop complaining about it..